compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -198,30 +198,64 @@ outlineDispatchExternOp(std::string name,
   auto parentFuncOp = dispatchExternOp->getParentOfType<FunctionOpInterface>();
   auto parentModuleOp = parentFuncOp->getParentOfType<mlir::ModuleOp>();
   OpBuilder parentModuleBuilder(&parentModuleOp.getBody()->back());
-  auto executableOp = parentModuleBuilder.create<IREE::Flow::ExecutableOp>(
+  auto executableOp = parentModuleBuilder.create<IREE::HAL::ExecutableOp>(
```

Comment:
There is a comment a few lines up that this leaves outdated

```
// Outlines a dispatch region into a flow.executable and replaces the region op
// with a dispatch to that outlined executable.
```

I'm also a bit worried to see the dialects mix more and more in these changes. This one in particular is strange, since the concept of an "executable" changes as lowering progresses, so each dialect defines its own version. What is the longer term plan for the layering? Should `Flow::ExecutableOp` expand in scope, or are more parts of the pipeline going to use `HAL` directly? Is it still useful to have a `Flow::ExecutableOp` if it isn't expressive enough?

Probably fine to proceed without fully resolving, but I don't quite have a mental model for how much cleanup work we're deferring. Do you expect to take a higher level view of Flow/Stream/HAL once the current sprint towards multi-targeting has achieved its goals?

Reply:
Mental model should be pretty simple: it's always valid to include HAL executables in inputs and the reason to have flow/stream ones are because we manipulate them in flow/stream. If you don't need to manipulate them - such as with external executables or ones you've crafted yourself (via triton/etc) then you don't need to use flow/stream executable ops. flow.dispatch/stream.async.dispatch/stream.cmd.dispatch can all dispatch any executable.

Reply:
Ah, good points! Thanks.